syntax = "proto3";

package Drunc;

service CommandProcessor {
  rpc execute_command (Command) returns (stream CommandResponse) {}
}

service PingProcessor {
  rpc ping (Ping) returns (stream Ping) {}
}

message Ping {
  string controlled_name = 1;
  string controller_name = 2;
  string datetime = 3;
}

message Command {
  string command_name = 1;
  string command_data = 2;
  string controlled_name = 3;
  string controller_name = 4;
  string datetime = 5;
}

message CommandResponse {
  enum ResponseCode {
    ACK = 0;
    DONE = 1;
    FAILED = 2;
  };
  
  ResponseCode response_code = 1;
  string response_text = 2;
  string command_name = 3;
  string command_data = 4;
  string controlled_name = 5;
  string controller_name = 6;
  string datetime = 7;
}
